umh-core/pkg/service/s6/s6.go

```diff
@@ -1192,6 +1213,17 @@ func (s *DefaultService) GetLogs(ctx context.Context, servicePath string, fsServ
 		copy(out, st.logs[:st.head])
 	}
 
+	// filter the logs to only include logs since last deployment time
+	if !getLastDeployedTime(servicePath).IsZero() {
+		filtered := out[:0] // reuse slice to avoid allocation
+		for _, entry := range out {
```

Comment:
you could add here one optimization: if we once reach a good timestamp, we can simply re-use all the other entries as they are already time sorted and dont have check all the other timestamps